src/documentation/system_administrators/advanced/ecommerce/nopcommerce.md

```diff
@@ -0,0 +1,269 @@
+- [Install nopCommerce](#install-nopcommerce)
+- [Access the Ecommerce from the Public Internet](#access-the-ecommerce-from-the-public-internet)
+  - [Set a DNS Record](#set-a-dns-record)
+  - [Access the Ecommerce](#access-the-ecommerce)
+  - [HTTPS with Caddy](#https-with-caddy)
+    - [Manage with Systemd](#manage-with-systemd)
+- [Access Admin Panel](#access-admin-panel)
+- [Manage nopCommerce with Systemd](#manage-nopcommerce-with-systemd)
+- [References](#references)
+- [Questions and Feedback](#questions-and-feedback)
+---
+
+## Introduction
+
+We show how to deploy a free and open-source ecommerce on the ThreeFold Grid. We will be deploying on a full VM with an IPv4 address.
+
+[nopCommerce](https://www.nopcommerce.com/en) is an open-source ecommerce platform based on Microsoft's ASP.NET Core framework and MS SQL Server 2012 (or higher) backend Database. 
+
+## Prerequisites
+
+- [A TFChain account](../../../dashboard/wallet_connector.md)
+- TFT in your TFChain account
+  - [Buy TFT](../../../threefold_token/buy_sell_tft/buy_sell_tft.md)
+  - [Send TFT to TFChain](../../../threefold_token/tft_bridges/tfchain_stellar_bridge.md)
+
+## Deploy a Full VM
+
+We start by deploying a full VM on the ThreeFold Dashboard.
+
+* On the [Threefold Dashboard](https://dashboard.grid.tf/#/), go to the [full virtual machine deployment page](https://dashboard.grid.tf/#/deploy/virtual-machines/full-virtual-machine/)
+* Deploy a full VM (Ubuntu 22.04) with an IPv4 address and at least the minimum specs for a full VM
+  * IPv4 Address
+  * Minimum vcores: 1vcore
+  * Minimum MB of RAM: 512MB
+  * Minimum storage: 15GB
+* After deployment, note the VM IPv4 address
+
+## Create an SSH Tunnel
+
+We create an SSH tunnel with port 5432:80, as it is this combination that we will set for nopCommerce on the docker-compose file.
+
+- Open a terminal and create an SSH tunnel
+    ```
+    ssh -4 -L 5432:127.0.0.1:80 root@VM_IPv4_address>
+    ```
+
+Simply leave this window open and follow the next steps.
+
+## Preparing the VM
+
+We prepare the full to run nopCommerce.
+
+* Connect to the VM via SSH
+    ``` 
+    ssh root@VM_IPv4_address
+    ```
+* Update the VM
+   ```
+   apt update
+   ```
+* [Install Docker](../../computer_it_basics/docker_basics.html#install-docker-desktop-and-docker-engine)
+* Install docker-compose
+    ```
+    apt install docker-compose -y
+    ```
+
+## Set a Firewall
+
+You can set a firewall to your VM for further security. This should be used in production mode.
+
+* Add the permissions
+  * ```
+    ufw allow 80
+    ufw allow 443
+    ```
+* Enable the firewall
+  * ```
+    ufw enable
+    ```
+* Verify the fire wall status
+  * ```
+    ufw status verbose
+    ```
+
+## Download nopCommerce
+
+* Clone the repository
+   ```
+   git clone https://github.com/nopSolutions/nopCommerce.git
+   cd nopCommerce
+   ```
+* Build the image
+   ```
+   cd nopCommerce
+   docker-compose -f ./postgresql-docker-compose.yml build
```

Comment:
followed the docs and every time the machine stops responding while bulding the docker image 
![Screenshot from 2024-05-15 13-41-01](https://github.com/threefoldtech/info_grid/assets/110984055/5c6352a5-b352-4e7a-a03c-f4461bad0a8e)
i tried 3 times. shall I try again with more cores than the documented ones ? @Mik-TF

Reply:
Oh that strange. Good idea indeed. I will try on my end too, just in case. It was smooth sailing on my end but perhaps I forgot something in the guide!

Reply:
@khaledyoussef24 I think you're right. I must have set at least 2 vcores, 4 gb mem, 50gb SSD when I did it first.

Reply:
I just tested with more resources and it works.

I also set ssh in ufw, otherwise the tunneling didn't work anymore. Good catch Khaled!

Reply:
perfect, it works fine with these changes.